### ISSUE TITLE
Integrate TransactionContext in Js tracer

### ISSUE THREAD

**Issue by mattsse** [paradigmxyz/reth#6088]
### Describe the feature

the JsTracer now supports additional tx related fields: https://github.com/paradigmxyz/evm-inspectors/blob/main/src/tracing/js/mod.rs#L547

which we need to provide here:

https://github.com/paradigmxyz/reth/blob/630fd45da6482a150ceedf612293c3cd6773a1dc/crates/rpc/rpc/src/debug.rs#L561-L561 

for this the https://github.com/paradigmxyz/reth/blob/630fd45da6482a150ceedf612293c3cd6773a1dc/crates/rpc/rpc/src/debug.rs#L491-L491

function likely needs to be changed to accept `Option<TransactionContext>`

**Comment by supernovahs** [comment]
I am trying to update the revm inspectors in my local 
but cargo update is throwing error due to alloy types i think
```
cargo update revm-inspectors
    Updating git repository `https://github.com/paradigmxyz/evm-inspectors`
    Updating crates.io index
    Updating git repository `https://github.com/bluealloy/revm`
    Updating git repository `https://github.com/alloy-rs/alloy`
error: failed to select a version for `alloy-rpc-trace-types`.
    ... required by package `reth-rpc-types v0.1.0-alpha.14 (/Users/supernovahs/Desktop/reth/crates/rpc/rpc-types)`
    ... which satisfies path dependency `reth-rpc-types` (locked to 0.1.0-alpha.14) of package `reth-primitives v0.1.0-alpha.14 (/Users/supernovahs/Desktop/reth/crates/primitives)`
    ... which satisfies path dependency `reth-primitives` (locked to 0.1.0-alpha.14) of package `reth-net-common v0.1.0-alpha.14 (/Users/supernovahs/Desktop/reth/crates/net/common)`
versions that meet the requirements `*` are: 0.1.0

the package `reth-rpc-types` depends on `alloy-rpc-trace-types`, with features: `jsonrpsee-types` but `alloy-rpc-trace-types` does not have these features.


failed to select a version for `alloy-rpc-trace-types` which could resolve this conflict

```

**Comment by mattsse** [comment]
fixed #6093

**Comment by supernovahs** [comment]
@mattsse 
```
error: failed to select a version for `alloy-rpc-types`.
    ... required by package `reth-rpc-types v0.1.0-alpha.14 (/Users/supernovahs/Desktop/reth/crates/rpc/rpc-types)`
    ... which satisfies path dependency `reth-rpc-types` of package `reth-primitives v0.1.0-alpha.14 (/Users/supernovahs/Desktop/reth/crates/primitives)`
    ... which satisfies path dependency `reth-primitives` of package `reth-net-common v0.1.0-alpha.14 (/Users/supernovahs/Desktop/reth/crates/net/common)`
versions that meet the requirements `*` are: 0.1.0

the package `reth-rpc-types` depends on `alloy-rpc-types`, with features: `ssz` but `alloy-rpc-types` does not have these features.


failed to select a version for `alloy-rpc-types` which could resolve this conflict

```
